docs/genesis-preparation.md

```diff
@@ -0,0 +1,352 @@
+# Genesis Preparation
+
+This article describes the steps should be taken to prepare genesis block.
+
+## 1. Nodes Keys
+
+First of all, secret/public keys for the cluster's nodes should be generated. Nodes will
+use them to perform blockchain-related procedures (for examples, to sign new blocks).
+These keys should be uploaded to the nodes during cluster deployment.
+
+### 1.1. Secret Keys
+
+To generate secret keys for nodes use this command:
+
+```
+$ stack exec -- cardano-keygen generate-key --path PATH
```

Comment:
I attempted this on `avieth/csl1617_config`, but I get the following error:

```
ConfigurationOptions {cfoFilePath = "node/configuration.yaml", cfoKey = "default", cfoSystemStart = Nothing}
cardano-keygen: MissingSystemStartTime
```

There's a `--system-start` option. The docs should explicitly mention that its value needs to be decided by this point, and call out who's responsible for setting it.